x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TransportRollupAction.java

```diff
@@ -558,6 +558,10 @@ private void createRollupIndex(
                 .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
                 .put(IndexSettings.INDEX_REFRESH_INTERVAL_SETTING.getKey(), "-1")
                 .put(IndexMetadata.INDEX_DOWNSAMPLE_STATUS.getKey(), IndexMetadata.DownsampleTaskStatus.STARTED)
+                .put(
+                    MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.getKey(),
+                    sourceIndexMetadata.getSettings().get(MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.getKey())
+                )
```

Comment:
Will this work if `MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING` is null at the source index? 
Do we know what happens in this case?

Reply:
I can add a test for this maybe?

Reply:
Or better I can make the value random...